### PR TITLE
Kaa list bug

### PR DESCRIPTION
**Pull request by forGGe**
Appropriate test is included.

### CODE REVIEW

File: client/client-multi/client-c/src/kaa/collections/kaa_list.c [hunk 1]
```diff
@@ -302,7 +302,9 @@ static kaa_list_node_t *kaa_split_util( kaa_list_node_t *head)
 
 void kaa_list_sort(kaa_list_t *list, match_predicate pred)
 {
-    kaa_list_node_t *node = kaa_merge_sort(kaa_list_begin(list), pred);
+    kaa_list_node_t *head = kaa_list_begin(list);
+    KAA_RETURN_IF_NIL(head,);
```
Comment (ghost) [inline client/client-multi/client-c/src/kaa/collections/kaa_list.c]: How about to check a list object and its size instead?

**Comment by ghost** [comment]
Reviewed.

**Comment by rasendubi** [comment]
Is it appropriate to modify `kaa_merge_sort` to handle this case instead?

**Comment by rasendubi** [comment]
Disscussed. LGTM

**Comment by ghost** [comment]
LGTM

**Pull request merged**